### ISSUE TITLE
opam lint --normalise should not include error messages to its stdout output.

### ISSUE THREAD

**Issue by camlspotter** [ocaml/opam#2962]
I found `opam lint --normalise` is useful to reformat my `opam` files but the output is contaminated with lint error and warning messages:

```
.../ppx_orakuda/opam/opam: Errors.
            error 21: Field 'opam-version' doesn't match the current version, validation may not be
              accurate: "1.2"
            error 32: Field 'ocaml-version:' and variable 'ocaml-version' are deprecated, use a
              dependency towards the 'ocaml' package instead for availability, and the 'ocaml:version'
              package variable for scripts
opam-version: "1.2"
version: "3.0.3"
homepage: "https://bitbucket.org/camlspotter/orakuda/"
bug-reports:
  "https://bitbucket.org/camlspotter/orakuda/issues?status=new&status=open"
depends: [
  "ocamlfind" {build}
  "omake" {build}
...
```

It makes `--normalise` lousy to use.  Is it possible to

* Output warnings and errors to stderr
* or specify the output file for the normalised file ?

Thanks.

**Comment by camlspotter** [comment]
The "Passed" message is also output to stdout: `~/.../opamfind/opam/opam: Passed.`

It seems the best solution is to add something like `-o <path>` option for the normalised output.

**Comment by dra27** [comment]
I'm not sure `-o` is the best way - I think your initial suggestion is correct - errors, warnings and status should go to stderr and the normalised output should go to stdout. If there's a compatibility concern (I've not checked whether opam lint sets exit status appropriately) then perhaps only `--normalise` would trigger the new behaviour?

**Comment by camlspotter** [comment]
I just worried the change from stdout to stderr might affect all the message outputs of opam command.  Personally I am ok with both of the solutions.

**Comment by dra27** [comment]
No - it's just three places in the code for the lint command ([here](https://github.com/ocaml/opam/blob/master/src/client/opamCommands.ml#L2493) and two places following) which need a bit more logic to call the correct OpamConsole function rather than `msg`.

**Comment by AltGr** [comment]
Yes, I think it is a bug, these messages should be to stderr.

**Comment by AltGr** [comment]
Note that, for just reformatting files, it is preferred to use `opam show --file`. I am not sure wether this was already in `1.2.2` or not though.

**Comment by camlspotter** [comment]
`opam show --file` is not available in `1.2.2`.
Thanks for the fix!
